core/dbt/events/functions.py

```diff
@@ -138,25 +137,21 @@ def event_to_serializable_dict(
     ts_fn: Callable[[datetime], str]
 ) -> Dict[str, Any]:
     data = dict()
-    node_info = dict()
+    node_info: Dict[str, Any] = dict()
     log_line = dict()
     try:
-        log_line = dataclasses.asdict(e, dict_factory=type(e).asdict)
-    except AttributeError:
+        log_line = e.to_dict()
+    except AttributeError as exc:
         event_type = type(e).__name__
         raise Exception(  # TODO this may hang async threads
-            f"type {event_type} is not serializable to json."
-            f" First make sure that the call sites for {event_type} match the type hints"
-            f" and if they do, you can override the dataclass method `asdict` in {event_type} in"
-            " types.py to define your own serialization function to a dictionary of valid json"
-            " types"
+            f"type {event_type} is not serializable. {str(exc)}"
```

Comment:
When do we expect this exception to be hit with the new implementation? What do we do to convert something that's not serializable to something that is inside the event module?